amqp-impl/src/main/java/io/streamnative/pulsar/handlers/amqp/AmqpProtocolHandler.java

```diff
@@ -162,4 +162,17 @@ public static int getListenerPort(String listener) {
         int lastIndex = listener.lastIndexOf(':');
         return Integer.parseInt(listener.substring(lastIndex + 1));
     }
+
+    public static String getAppliedAmqpListeners(AmqpServiceConfiguration configuration) {
+        String amqpListeners = configuration.getAmqpListeners();
+        if (amqpListeners == null) {
+            String fullyHostName = ServiceConfigurationUtils.getDefaultOrConfiguredAddress(amqpListeners);
```

Comment:
If the `amqpListeners` is null, we could use the method `ServiceConfigurationUtils.unsafeLocalhostResolve` directly.